test/authController.test.js

```diff
@@ -44,18 +51,27 @@ afterAll(async () => {
   // turn off the order update interval so that jest can cleanly shutdown
   clearInterval(orderUpdateInterval);
 });
+afterEach(async () => {
+  await User.deleteMany({});
+  jest.restoreAllMocks();
+}, 15000);
 
 describe('Authentication API Endpoints', () => {
+  beforeEach(async () => {
+    await User.create(testUser);
+  });
```

Comment:
Now we have **two** `beforeEach` blocks that are **both** trying to create the same test user, before each test. One on line 43-45 and one on line 60-62.

The one on line 43-45 is _outside_ any `describe` block, so runs _before every test in the file._  
The one on line 60-62 is _inside_ a `describe` block so it runs _before every test in that describe block._

You'll see in the test logs (if you run the test locally or if you check in the github run that right before the `POST /api/v1/auth/register` it logs the error "Email already exists": [https://github.com/Code-the-Dream-School/dd-prac-team4-back/actions/runs/6508303940/job/17677395210](https://github.com/Code-the-Dream-School/dd-prac-team4-back/actions/runs/6508303940/job/17677395210)

This is caused by trying to create a user with the same email twice. 

We can remove one of these `beforeEach` blocks; in this case it doesn't really matter which one we remove since we only have one `describe` block. But likely it's better to remove the one on line 60-62 so that if someone does eventually add more \`describe \`blocks to this test file, the test user will continue to be created for those separate `describe` blocks as well.

Reply:
I have removed the second beforeEach block but it doesn't work, it gives me the same error 
<img width="916" alt="image" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/88336329/9869dde6-25bf-42b2-b11e-cc9c25461161">

Reply:
Yep, keep in mind that when the user in the test tries to register, it's again going to try to make a user with that email when they call the `/register` endpoint: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/pull/76#discussion_r1353487977](https://github.com/Code-the-Dream-School/dd-prac-team4-back/pull/76#discussion_r1353487977)

> As for `register` , of course the same user cannot register if they have already been created, so in that test we would need to delete Ava from the db at the beginning of the test before proceeding.

So at the beginning of the register test only, you'll need to delete the existing Ava user: `await User.deleteMany({})` so that they can register via the endpoint successfully.

Reply:
Ok now I understood, thank you 👍